tests/testscript.py

```diff
@@ -10,7 +10,46 @@
     from nixos_test_stubs import start_all, computeVM, controllerVM  # type: ignore
 
 
-class LibvirtTests(unittest.TestCase):
+class PrintLogsOnErrorTestCase(unittest.TestCase):
+    """
+    Custom TestCase class that prints interesting logs in error case.
+    """
+    def run(self, result=None):
+        if result is None:
+            result = self.defaultTestResult()
+
+        original_addError = result.addError
```

Comment:
what is the difference between error and failure here?

Reply:
The code overwrites the original callbacks with custom ones, while calling the original callbacks from the custom callback.

I don't know what the difference between error and failure is. Event looking through the unittest source code does not help here, it seems the terms are even mixed in some function documentation. 

It may be that there can be tests that are expected to fail, and therefore they make a difference.